### PR TITLE
Genomic files tab

### PR DESCRIPTION
**Pull request by xlecours**
Altered Profile tab
new Files tab
1 modality upload

**Release notes**: This requires a genomic_uploder directory under the GenomicDataPath (Config) directory. this directory needs to be r+w for the apache user.

### CODE REVIEW

File: SQL/2016-04-02-GenomicBrowser-FileUploader.sql [hunk 1]
```diff
@@ -0,0 +1,29 @@
+CREATE TABLE `genomic_analysis_modality_enum` (
+  `analysis_modality` varchar(100),
+  PRIMARY KEY (`analysis_modality`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8
+COMMENT '';
+
+INSERT IGNORE INTO `genomic_analysis_modality_enum` (analysis_modality) VALUES
+('Methylation beta-values'),
+('Other');
```
Comment (christinerogers) [inline SQL/2016-04-02-GenomicBrowser-FileUploader.sql]: Let's add SNP, CNV and GWAS in here, if it makes sense.  To have ready options pre-populated, and also to suggest to developers what this is for.

File: modules/genomic_browser/README.md [hunk 2]
```diff
@@ -4,6 +4,24 @@
 
 To use the methylation tab, the genomic_cpg_annotaion table should be filled with data from the Illumina HumanMethylation450k probe annotations file. This file can be found at ftp://webdata2:webdata2@ussd-ftp.illumina.com/downloads/ProductFiles/HumanMethylation450/HumanMethylation450_15017482_v1-2.csv. We provide a script,  `modules/genomic_browser/tools/HumanMethylation450k_annotations_to_sql` to transforme the csv file from Illumina's FTP to into a mysql transaction file. The output of the tool can be piped to mysql. The process should take between 5 to 10 minutes.
```
Comment (christinerogers) [inline modules/genomic_browser/README.md]: Fantastic.  Add one note that this requires python which is not covered in Loris dependencies by default. (can do this after today)

File: modules/genomic_browser/ajax/genomic_file_upload.php [hunk 3]
```diff
@@ -0,0 +1,471 @@
+<?php
+/**
+ * This allow to upload file for the genomic module.
```
Comment (christinerogers) [inline modules/genomic_browser/ajax/genomic_file_upload.php]: allows uploading of files for...

File: modules/genomic_browser/ajax/genomic_file_upload.php [hunk 4]
```diff
@@ -0,0 +1,471 @@
...
+ *
+ * NOTE : This require a genomic_uploader directory under the
+ * GenomicDataPath directory specified in the cconfigSettings.
+ * This directory needs to be accessible w/r to the apache user.
```
Comment (christinerogers) [inline modules/genomic_browser/ajax/genomic_file_upload.php]: This Note paragraph - fantastic.  It should (also?) go in the Readme. 

little things: cconfigSettings;  require->requires, spell out w/r (with respect)
accessible -> executable

File: modules/genomic_browser/ajax/genomic_file_upload.php [hunk 5]
```diff
@@ -0,0 +1,471 @@
...
+ *  @package    Genomic_Module
+ *  @author     Loris Team <loris.mni@bic.mni.mcgill.ca>
+ *  @contriutor Xavier Lecours boucher <xavier.lecoursboucher@mcgill.ca>
+ *  @license    Loris license
+ *  @link       https://github.com/aces/Loris-Trunk
+ */
+
+$userSingleton =& User::singleton();
+if (!$userSingleton->hasPermission('genomic_browser_view_site')
+    && !$userSingleton->hasPermission('genomic_browser_view_allsites')
```
Comment (christinerogers) [inline modules/genomic_browser/ajax/genomic_file_upload.php]: Adding an upload permission.  Can we add this for 16.1 or better for 16.0?

File: modules/genomic_browser/README.md [hunk 6]
```diff
@@ -4,6 +4,24 @@
...
+usage :
+```
 python3 HumanMethylation450k_annotations_to_sql.py <annotation_file> | mysql -u <user> -p <database>
 ```
+
+## Files
+
+
+To enable uploading of large files via the genomic file uploader, update apache configurations to increase size and time limits.  (Sample values suggested below)
+
```
Comment (christinerogers) [inline modules/genomic_browser/README.md]: Also copy here the "Note: " paragraph from in ajax/*.php file below
Add: It's recommended that this configSetting should probably be soft-linked to /data/ path

File: modules/genomic_browser/jsx/FileUploadModal.jsx [hunk 7]
```diff
@@ -0,0 +1,332 @@
...
+
+    getInitialState: function () {
+        return {
+            baseURL: '',
+            fileType: "",
+            useColumnHeaders: true, // Change this to false when we are ready to use Mapping files
+        };
+    },
+
+    getDeafaultProps: function () {
```
Comment (christinerogers) [inline modules/genomic_browser/jsx/FileUploadModal.jsx]: Default (typo here, inconsistently named)

File: modules/genomic_browser/js/FileUploadModal.js [hunk 8]
```diff
@@ -0,0 +1,414 @@
...
+
+    getInitialState: function () {
+        return {
+            baseURL: '',
+            fileType: "",
+            useColumnHeaders: true };
+    },
+
+    // Change this to false when we are ready to use Mapping files
+    getDeafaultProps: function () {
```
Comment (christinerogers) [inline modules/genomic_browser/js/FileUploadModal.js]: Default (typo here, inconsistently named)

File: SQL/2016-04-02-GenomicBrowser-FileUploader.sql (hunk 1 above)
Comment (xlecours) [inline SQL/2016-04-02-GenomicBrowser-FileUploader.sql]: This is an enum. 
The uploader will show every 'modalities' in that table

File: modules/genomic_browser/ajax/genomic_file_upload.php (hunk 4 above)
Comment (xlecours) [inline modules/genomic_browser/ajax/genomic_file_upload.php]: done

File: modules/genomic_browser/ajax/genomic_file_upload.php (hunk 3 above)
Comment (xlecours) [inline modules/genomic_browser/ajax/genomic_file_upload.php]: done

File: modules/genomic_browser/ajax/genomic_file_upload.php (hunk 5 above)
Comment (xlecours) [inline modules/genomic_browser/ajax/genomic_file_upload.php]: I will be a different pull request if this one is merge.

File: modules/genomic_browser/README.md (hunk 6 above)
Comment (xlecours) [inline modules/genomic_browser/README.md]: done

File: modules/genomic_browser/jsx/FileUploadModal.jsx (hunk 7 above)
Comment (xlecours) [inline modules/genomic_browser/jsx/FileUploadModal.jsx]: Done

File: modules/genomic_browser/js/FileUploadModal.js (hunk 8 above)
Comment (xlecours) [inline modules/genomic_browser/js/FileUploadModal.js]: Done

File: modules/genomic_browser/jsx/FileUploadModal.jsx [hunk 9]
```diff
@@ -0,0 +1,332 @@
...
+                    default:
+                        console.log('?');
+                        break;
+                }
+            }
+            catch (e){
+                console.error("[XHR STATECHANGE] Exception: " + e);
+            }
+        };
+        var url = this.props.baseURL + "/AjaxHelper.php?Module=genomic_browser&script=genomic_file_upload.php";
```
Comment (driusan) [inline modules/genomic_browser/jsx/FileUploadModal.jsx]: This should use the pretty URL

**Pull request merged**